server/cluster/coordinator.go

```diff
@@ -5,25 +5,65 @@ package cluster
 import (
 	"context"
 	"sync"
+	"time"
 
 	"github.com/CeresDB/ceresdbproto/pkg/clusterpb"
+	"github.com/CeresDB/ceresmeta/pkg/log"
+	"github.com/CeresDB/ceresmeta/server/schedule"
 	"github.com/pkg/errors"
+	"go.uber.org/zap"
 )
 
+// TODO: heartbeatInterval should be set in config
+const DefaultHeartbeatInterval = time.Second * 3
+
 // coordinator is used to decide if the shards need to be scheduled.
 type coordinator struct {
 	// Mutex is to ensure only one coordinator can run at the same time.
-	sync.Mutex
+	lock    sync.Mutex
 	cluster *Cluster
+
+	ctx          context.Context
+	cancel       context.CancelFunc
+	bgJobWg      *sync.WaitGroup
+	eventHandler *schedule.EventHandler
 }
 
-func (c *coordinator) Run(ctx context.Context) error {
-	c.Lock()
-	defer c.Unlock()
-	if err := c.scatterShard(ctx); err != nil {
-		return errors.Wrap(err, "coordinator Run")
+func newCoordinator(cluster *Cluster, hbstream *schedule.HeartbeatStreams) *coordinator {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &coordinator{
+		cluster: cluster,
+
+		ctx:          ctx,
+		cancel:       cancel,
+		eventHandler: schedule.NewEventHandler(hbstream),
 	}
-	return nil
+}
+
+func (c *coordinator) runBgJob() {
```

Comment:
Provide a concrete name (e.g. dispatchNoneEventPeriodically) if no more jobs are run in background.

Reply:
`dispatchNoneEventPeriodically` is one job, maybe more schedulers run background.